auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -572,22 +575,40 @@ public PhoneBuilder setDefaultPhoneNumber(String number) {
              * @param number the phone number in local format
              */
             @NonNull
-            public PhoneBuilder setDefaultPhoneNumber(String code, String number) {
+            public PhoneBuilder setDefaultPhoneNumber(int code, @NonNull String number) {
                 if (getParams().containsKey(ExtraConstants.EXTRA_PHONE)) {
                     throw new IllegalStateException("Cannot overwrite previously set phone number");
                 }
 
-                getParams().putString(ExtraConstants.EXTRA_COUNTRY_CODE, code);
+                getParams().putString(ExtraConstants.EXTRA_COUNTRY_CODE, String.valueOf(code));
                 getParams().putString(ExtraConstants.EXTRA_NATIONAL_NUMBER, number);
 
                 return this;
             }
+
+            @NonNull
+            public PhoneBuilder setDefaultCountryCode(int code) {
+                if (getParams().containsKey(ExtraConstants.EXTRA_PHONE)
+                        || getParams().containsKey(ExtraConstants.EXTRA_COUNTRY_CODE)
+                        || getParams().containsKey(ExtraConstants.EXTRA_NATIONAL_NUMBER)) {
+                    throw new IllegalStateException("Cannot overwrite previously set phone number");
+                }
+
+                getParams().putString(ExtraConstants.EXTRA_COUNTRY_CODE, String.valueOf(code));
+
+                return this;
+            }
         }
 
         public static final class GoogleBuilder extends Builder {
             public GoogleBuilder() {
                 //noinspection deprecation taking a hit for the backcompat team
                 super(GoogleAuthProvider.PROVIDER_ID);
+                if (getApplicationContext().getString(R.string.default_web_client_id)
+                        .equals(UNCONFIGURED_CONFIG_VALUE)) {
+                    throw new RuntimeException("Check your google-services plugin configuration," +
```

Comment:
What about extracting hardcoded string values like this to strings.xml?

Reply:
See https://github.com/firebase/FirebaseUI-Android/pull/1083#discussion_r159147678, nope.